src/lib.rs

```diff
@@ -618,23 +744,43 @@ impl fmt::Display for SetLoggerError {
     }
 }
 
+// The Error trait is not available in libcore
+#[cfg(feature = "use_std")]
 impl error::Error for SetLoggerError {
     fn description(&self) -> &str { "set_logger() called multiple times" }
 }
 
+/// The type returned by `shutdown_logger_raw` if `shutdown_logger_raw` has
+/// already been called or if `set_logger_raw` has not been called yet.
+#[allow(missing_copy_implementations)]
+#[derive(Debug)]
+pub struct ShutdownLoggerError(());
+
+impl fmt::Display for ShutdownLoggerError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "attempted to shut down a logger without an active logger")
+    }
+}
+
+// The Error trait is not available in libcore
+#[cfg(feature = "use_std")]
+impl error::Error for ShutdownLoggerError {
```

Comment:
@alexcrichton there's nothing blocking us from adding `Error` to libcore right?

Doesn't affect this PR of course.